pkg/krmtotf/fetchlivestate.go

```diff
@@ -72,8 +69,26 @@ func hasEmptySelfLink(resource *Resource) bool {
 	return false
 }
 
+// SkipParentReadyCheckForDeletion Skip the parent ready check for allowlist resources,
+// when parent exists but has deletion failed error.
+// Due to their API design, the resource is deletable even parent is not ready.
+// See b/306583728#comment8 for details.
+func SkipParentReadyCheckForDeletion(resource *Resource, parent *k8s.Resource) bool {
```

Comment:
Since we are doing special handling for certain resources. as a helper function, I feel it's straight-forward to return true when the condition meets. I would prefer to negate the condition when we handle it, i.e.
if (!condition) {//do something}
instead of negate it in the helper function.
I also found some similar usage in the code base: [example](https://source.corp.google.com/piper///depot/google3/net/rpc/go/server.go;bpv=1;bpt=1;rcl=599940295;l=3043?q=%22func%20skip%22%20f:.go&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental). 
I feel this is nothing about correctness but readability, open for suggestions.

Reply:
Updated the function name and revised the code a bit.